roles/keystone/tasks/main.yml

```diff
@@ -72,12 +72,19 @@
 
 - name: configure keystone
   template: src={{ item }} dest=/etc/keystone/
-            mode={{ 0644 if 'policy.json' in item else 0640 }}
+            mode=0640 
             owner=keystone group=keystone
   with_fileglob: ../templates/etc/keystone/*
   notify:
     - restart keystone services
 
+- name: configure keystone policy
+  template: src=roles/keystone-defaults/templates/etc/keystone/policy.json dest=/etc/keystone/policy.json
```

Comment:
You shouldn't have to use the path of roles/keystone-defaults/templates/, if you just reference the template by relative name, it'll look in the templates/ directories of every parent role.

Same goes for the reference in the horizon task.